mlflow/utils/environment.py

```diff
@@ -80,5 +85,40 @@ def _get_additional_pip_dep(conda_env):
 
 
 def _log_pip_requirements(conda_env, path, requirements_file="requirements.txt"):
-    pip_deps = _get_additional_pip_dep(conda_env)
+    pip_deps = _get_pip_dep(conda_env)
     _mlflow_additional_pip_env(pip_deps, path=os.path.join(path, requirements_file))
+
+
+def _resolve_pip_requirements(pip_requirements):
+    if isinstance(pip_requirements, str):
+        return [
+            req.requirement for req in parse_requirements(pip_requirements, session=PipSession())
+        ]
+    elif isinstance(pip_requirements, list):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmp_path = os.path.join(tmpdir, f"{uuid.uuid4().hex}.txt")
+            with open(tmp_path, "w") as f:
+                f.write("\n".join(pip_requirements))
+            return [req.requirement for req in parse_requirements(tmp_path, session=PipSession())]
```

Comment:
I think we can directly resolve a list of packages rather than putting them in a tempoary text file, but it requires us to import more internal functions.